src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
```

Comment:
Since Node is going to be an inner class, you won't need getters and setters for its fields. You can access the fields directly within the MyLinkedList class.